src/test/java/com/GotYourBac/gotYourBac/GotYourBacApplicationTests.java

```diff
@@ -1,13 +1,43 @@
 package com.GotYourBac.gotYourBac;
 
+import com.GotYourBac.gotYourBac.controllers.UserController;
+import com.GotYourBac.gotYourBac.models.ApplicationUserRepository;
 import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 
 @SpringBootTest
+@AutoConfigureMockMvc
 class GotYourBacApplicationTests {
 
+	@Autowired
+	MockMvc mockMvc;
+
+	@Autowired
+	UserController userController;
+
 	@Test
 	void contextLoads() {
+//		assertThat();
+	}
+
+	//Test homepage rendering
+	@Test
+	public void shouldHaveHomePage() throws Exception {
+		this.mockMvc.perform(get("/"))
+				.andDo(print())
+				.andExpect(status().isOk())
+				.andExpect(content().string("<h2>Make new users here</h2>"));
```

Comment:
this test is already a failing test, you need to write tests that reference the files you have written. there is no h2 that says "make new users here"

Reply:
tests are passing now. I will add more tests for POST requests tomorrow